src/data_morph/data/dataset.py

```diff
@@ -52,6 +52,9 @@ def __init__(
         self.df: pd.DataFrame = self._validate_data(df).pipe(self._scale_data, scale)
         """pandas.DataFrame: DataFrame containing columns x and y."""
 
+        self._x = self.df['x'].to_numpy()
+        self._y = self.df['y'].to_numpy()
```

Comment:
Should these be properties? If we change the DataFrame, these will no longer match.